packages/transport/src/modules/tokens/TokenController.ts

```diff
@@ -176,22 +178,24 @@ export class TokenController extends TransportController {
             expiresAt: CoreDate.from(response.expiresAt),
             createdBy: CoreAddress.from(response.createdBy),
             createdByDevice: CoreId.from(response.createdByDevice),
-            content: plaintextTokenContent
+            content: plaintextTokenContent,
+            forIdentity: response.forIdentity ? CoreAddress.from(response.forIdentity) : undefined
         });
         return cachedToken;
     }
 
     public async loadPeerTokenByTruncated(truncated: string, ephemeral: boolean): Promise<Token> {
         const reference = TokenReference.fromTruncated(truncated);
-        return await this.loadPeerTokenByReference(reference, ephemeral);
+        return await this.loadPeerToken(reference.id, reference.key, ephemeral, reference.forIdentityTruncated);
     }
 
-    public async loadPeerTokenByReference(tokenReference: TokenReference, ephemeral: boolean): Promise<Token> {
-        return await this.loadPeerToken(tokenReference.id, tokenReference.key, ephemeral);
-    }
-
-    public async loadPeerToken(id: CoreId, secretKey: CryptoSecretKey, ephemeral: boolean): Promise<Token> {
+    private async loadPeerToken(id: CoreId, secretKey: CryptoSecretKey, ephemeral: boolean, forIdentityTruncated?: string): Promise<Token> {
         const tokenDoc = await this.tokens.read(id.toString());
+        if (!tokenDoc && forIdentityTruncated && !this.parent.identity.address.toString().endsWith(forIdentityTruncated)) {
+            // if you created the token, it exists already
```

Comment:
the comment in this line is misleading, I would 

a) completely remove it
b) put it before the if

I'd vote for a, but if you choose b please add a blank line between the comment and `const tokenDoc`